Keystone v2 and v3 manual and amulet/bundle testing

ajkavanagh opened the pull request:
This provides both keystone v2 and v3 setup for testing barbican
manually.  It demonstrates how to set up a project/tenant in each v3 and
v2 and the how to store a secret.  Then only differences are in the
authentication with Keystone.  However, it does demonstrate both
working.

This change also contains the v2 and v3 keystone tests with barbican.
These are pesented as two gate-_-v2 and gate-_-v3 tests which activate
testing with keystone v2 and keystone v3.

ajkavanagh committed:
Keystone v2 and v3 manual and amulet/bundle testing

This provides both keystone v2 and v3 setup for testing barbican
manually.  It demonstrates how to set up a project/tenant in each v3 and
v2 and the how to store a secret.  Then only differences are in the
authentication with Keystone.  However, it does demonstrate both
working.

This change also contains the v2 and v3 keystone tests with barbican.
These are pesented as two gate-*-v2 and gate-*-v3 tests which activate
testing with keystone v2 and keystone v3.

ajkavanagh commented:
Still a WIP as just sorting out some final bits of testing.

ajkavanagh commented:
Ready to go now.

javacruft commented on src/test-requirements.txt:
```diff
@@ -0,0 +1,16 @@
+charm-tools>=2.0.0
+# amulet deployment helpers
+#bzr+lp:charm-helpers#egg=charmhelpers
+bzr+lp:~ajkavanagh/charm-helpers/add-v3-keystone-catalog-check#egg=charmhelpers
```
This will need updating before we can land this change.

gnuoy commented:
Please remove the manual_testing directory

Commit:
Add .gitreview and clean up repo

Add src/README.md placeholder to prevent inheritence of the
lower layer readme file in the built charm.

Add src/tox.ini and test-requirements for Amulet prep.

Update metadata.yaml for series.

Update tox.ini to minimize and clarify pep8 ignores.

Resolve flake8/pbr test-requirements version range conflict.

Remove legacy lint and test targets from Makefile, ensuring
these tests will execute via tox.  Trusty test runners have
no py35 interpreter, and would have failed in running both
py34 and py35 in succession as it was defined.

Update src/layer.yaml repo value to point to official location
and remove unnecessary options.

Change-Id: Ia663fda7761dca6dc08583314f00beedf1db60cc
